### DISCUSSION TITLE
Can we export application logs to s3?

### DISCUSSION THREAD

**Discussion by anjulaendowus** [open-telemetry/opentelemetry-java#6113] [Q&A]
We have a requirement where we need to export application logs to s3. One of the possible option would be to use fluentd and we are looking at the OTel collector as well. Please provide your suggestions on this.

**Comment by jack-berg** [comment]
OpenTelemetry logs supports log forwarding, where application logs are forwarded from the console using something like fluentd, or exporting logs directly to a collector, where application logs are exported to a collector or other storage directly from the application using a network protocol like OTLP. These solutions are described in more detail [here](https://opentelemetry.io/docs/instrumentation/java/manual/#logs) and [here](https://opentelemetry.io/docs/specs/otel/logs/#legacy-first-party-applications-logs).

Note that for the direct to collector approach, `opentelemetry-java` is bundled with an OTLP exporter (`OtlpHttpLogRecordExporter` / `OtlpGrpcLogRecordExporter`), but its also possible to implement your own `LogRecordExporter`.

**Reply by anjulaendowus** [reply]
So if we use the direct to collector approach, are we able to store our logs in S3 in parquet format?

**Reply by jack-berg** [reply]
You'll have to explore the options available in the collector. The collector has an extensible architecture which allows you to write your own receivers, processors, exporters, etc, but has a number bundled in (the set of components bundled will vary based on the distribution). Not sure if any of the community-provided exporters support s3 in parquest format: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter

**Reply by trask** [reply]
~this would be a better question for https://github.com/open-telemetry/opentelemetry-collector-contrib or [#otel-collector](https://cloud-native.slack.com/archives/C01N6P7KR6W) slack channel~ (see @jack-berg's response above)
